redesign analytics email

rolandpo opened the pull request:
add new analytics fields to weekly and monthly emails and style according to design

some questions:
- are the analytics reports going to contain the same information as the emails or do they expand on the data?
- does the 'view more' button in the design show some additional transaction data or does it link somewhere?

![image](https://user-images.githubusercontent.com/62400812/212351189-b52da273-87fd-4aad-9bca-aaabfbc629b8.png)


![image](https://user-images.githubusercontent.com/62400812/212350020-2126d7b5-c69c-4444-b82f-4640b77f3f0e.png)

shahthepro commented on eagleproject/core/blockchain/harvest/transaction_history.py:
```diff
@@ -252,11 +285,14 @@ def calculate_difference(current_stat, previous_stat):
     supply_data = json_report["supply_data"] if "supply_data" in json_report else None
     supply_data_previous = json_report_previous["supply_data"] if "supply_data" in json_report_previous else None
 
+    ogv_data = json_report["ogv_data"] if "ogv_data" in json_report else None
```
nit: Assuming that `json_report` is a dict, why not use `json_report.get("ogv_data")`? The `.get()` method returns the value of the key if it exists or `None` otherwise

rolandpo replied:
I think json_report is a list

shahthepro commented on eagleproject/core/blockchain/harvest/transaction_history.py:
```diff
@@ -529,6 +575,26 @@ def fetch_supply_data(block_number):
 
+def fetch_ogv_data(to_block, from_timestamp, to_timestamp):
+    ogv_history = get_coin_history('OGV', from_timestamp, to_timestamp)
+    index = len(ogv_history['prices']) - 1
```
nit: what happens if `len(ogv_history)` is 0? I'm wondering if this whole block of code should be wrapped in a try catch?

rolandpo replied:
added a try catch block

shahthepro commented on eagleproject/core/coingecko.py:
```diff
@@ -5,10 +5,12 @@
 log = get_logger(__name__)
 
 COINGECKO_ENDPOINT = "https://api.coingecko.com/api/v3"
+DEFILLAMA_ENDPOINT = "https://stablecoins.llama.fi"
```
Would suggest to change the name of this file to something else, since it no longer has only Coingecko API code

rolandpo replied:
separated the files

shahthepro commented on eagleproject/core/templates/analytics_report_email.html:
```diff
@@ -1,150 +1,209 @@
-    <h2>{{type}} Analytics Report</h2>
-    <h4>
+  <div class="box">
+    <img src="/static/core/img/origin-dollar.svg" alt="origin dollar" class="wordmark" />
```
Should these be absolute URLs? I don't think relative URLs would work (unless Django does something when using a template)

shahthepro replied:
Also, even for absolute URLs, if the host of the images and files are different from the email domain, it might get flagged as spam/junk. It's probably worth testing these emails (using a SendGrid API key or something) before we merge it in

shahthepro commented on eagleproject/core/blockchain/harvest/transaction_history.py:
```diff
@@ -27,22 +27,23 @@
-from datetime import (
-    datetime,
-)
+import datetime
```
nit: Could probably do `from datetime import ( datetime, timedelta )` instead of importing everything from datatime and then using `datetime.datetime`

rolandpo replied:
added

shahthepro reviewed (commented):
Took a quick pass, looks good to me. I'll take another pass soon

rolandpo commented:
reworked the email template to use inline styling since most of it was being ignored after sending an email, added absolute image paths and switched out svgs for pngs for provider compatibility reasons

shahthepro reviewed (approved):
LGTM, Thanks for testing out the emails locally 🙏🏻
